client/packages/programs/src/api/api.ts

```diff
@@ -121,6 +122,19 @@ export const getEncounterQueries = (sdk: Sdk, storeId: string) => ({
       throw new Error('Could not find encounter');
     }
   },
+  byDocName: async (documentName: string): Promise<EncounterFragment> => {
+    const result = await sdk.encounterByDocName({ documentName, storeId });
+    const encounters = result?.encounters;
+
+    if (
+      encounters?.__typename === 'EncounterConnector' &&
+      !!encounters.nodes[0]
```

Comment:
Regarding the error you where seeing. You could fix it in a nicer way by returning an `Promise<EncounterFragment | undefined>` here and removing the `!!encounter.nodes[0]` check here.

To avoid two api call you can then disable the call by, e.g. in `useEncounterByDocName`, pass in `undefined` if the config only requires the program enrolment.

Reply:
just saw you already do the later part, this should have already fixed it?

Reply:
[0bcf79e](https://github.com/openmsupply/open-msupply/pull/1466/commits/0bcf79e2ca807825467d6bc842b7959f767c65a5)